tchannel/tchannel.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import (
+    absolute_import, division, print_function, unicode_literals
+)
+
+from tornado import gen
+
+from . import formats
+from .glossary import DEFAULT_TIMEOUT
+from .tornado import TChannel as DeprecatedTChannel
+
+__all__ = ['TChannel', 'Response']
+
+
+class TChannel(object):
+
+    def __init__(self, name, hostport=None, process_name=None,
+                 known_peers=None, formatters=None, trace=False):
+
+        # until we move everything here,
+        # lets compose the old tchannel
+        self._dep_tchannel = DeprecatedTChannel(
+            name, hostport, process_name, known_peers, trace
+        )
+
+        # if no formatters, use defaults
+        if not formatters:
+            formatters = formats.DEFAULT_FORMATS
+
+        # set formatters
+        for f in formatters:
+            f = f(self)
+            setattr(self, f.NAME, f)
+
+    @gen.coroutine
+    def call(self, format, service, arg1, arg2, arg3, timeout=None):
+
+        # TODO - dont use asserts for public API
+        assert format, "format is required"
+        assert service, "service is required"
+        assert arg1, "arg1 is required"
+
+        # default args
+        if arg2 is None:
+            arg2 = ""
+        if arg3 is None:
+            arg3 = ""
```

Comment:
seems like the `arg2` & `arg3` params should have `None` as the default in the method signature...